beacon-chain/core/electra/effective_balance_updates.go

```diff
@@ -0,0 +1,65 @@
+package electra
+
+import (
+	"fmt"
+
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	ethpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+)
+
+// ProcessEffectiveBalanceUpdates processes effective balance updates during epoch processing.
+//
+// Spec pseudocode definition:
+//
+//	def process_effective_balance_updates(state: BeaconState) -> None:
+//	    # Update effective balances with hysteresis
+//	    for index, validator in enumerate(state.validators):
+//	        balance = state.balances[index]
+//	        HYSTERESIS_INCREMENT = uint64(EFFECTIVE_BALANCE_INCREMENT // HYSTERESIS_QUOTIENT)
+//	        DOWNWARD_THRESHOLD = HYSTERESIS_INCREMENT * HYSTERESIS_DOWNWARD_MULTIPLIER
+//	        UPWARD_THRESHOLD = HYSTERESIS_INCREMENT * HYSTERESIS_UPWARD_MULTIPLIER
+//	        EFFECTIVE_BALANCE_LIMIT = (
+//	            MAX_EFFECTIVE_BALANCE_EIP7251 if has_compounding_withdrawal_credential(validator)
+//	            else MIN_ACTIVATION_BALANCE
+//	        )
+//
+//	        if (
+//	            balance + DOWNWARD_THRESHOLD < validator.effective_balance
+//	            or validator.effective_balance + UPWARD_THRESHOLD < balance
+//	        ):
+//	            validator.effective_balance = min(balance - balance % EFFECTIVE_BALANCE_INCREMENT, EFFECTIVE_BALANCE_LIMIT)
+func ProcessEffectiveBalanceUpdates(state state.BeaconState) error {
```

Comment:
If we pass the context we can be more inline with other process functions , and add a span to this for tracing.